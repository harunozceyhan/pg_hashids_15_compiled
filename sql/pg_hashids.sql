\set VERBOSITY terse
CREATE EXTENSION pg_hashids;

-- Encoding tests
SELECT id_encode(1001); -- Result: jNl
SELECT id_encode(1234567, 'This is my salt'); -- Result: Pdzxp
SELECT id_encode(1234567, 'This is my salt', 10); -- Result: PlRPdzxpR7
SELECT id_encode(1234567, 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 3GJ956J9B9

-- Decoding tests
SELECT id_decode('jNl'); -- Result: {1001}
SELECT id_decode('Pdzxp', 'This is my salt'); -- Result: {1234567}
SELECT id_decode('PlRPdzxpR7', 'This is my salt', 10); -- Result: {1234567}
SELECT id_decode('3GJ956J9B9', 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: {1234567}

SELECT id_decode_once('jNl'); -- Result: 1001
SELECT id_decode_once('Pdzxp', 'This is my salt'); -- Result: 1234567
SELECT id_decode_once('PlRPdzxpR7', 'This is my salt', 10); -- Result: 1234567
SELECT id_decode_once('3GJ956J9B9', 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 1234567

