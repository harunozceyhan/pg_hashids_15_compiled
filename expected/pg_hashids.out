\set VERBOSITY terse
CREATE EXTENSION pg_hashids;
-- Encoding tests
SELECT id_encode(1001); -- Result: jNl
 id_encode 
-----------
 jNl
(1 row)

SELECT id_encode(1234567, 'This is my salt'); -- Result: Pdzxp
 id_encode 
-----------
 Pdzxp
(1 row)

SELECT id_encode(1234567, 'This is my salt', 10); -- Result: PlRPdzxpR7
 id_encode  
------------
 PlRPdzxpR7
(1 row)

SELECT id_encode(1234567, 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 3GJ956J9B9
 id_encode  
------------
 3GJ956J9B9
(1 row)

-- Decoding tests
SELECT id_decode('jNl'); -- Result: {1001}
 id_decode 
-----------
 {1001}
(1 row)

SELECT id_decode('Pdzxp', 'This is my salt'); -- Result: {1234567}
 id_decode 
-----------
 {1234567}
(1 row)

SELECT id_decode('PlRPdzxpR7', 'This is my salt', 10); -- Result: {1234567}
 id_decode 
-----------
 {1234567}
(1 row)

SELECT id_decode('3GJ956J9B9', 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: {1234567}
 id_decode 
-----------
 {1234567}
(1 row)

SELECT id_decode_once('jNl'); -- Result: 1001
 id_decode_once 
----------------
           1001
(1 row)

SELECT id_decode_once('Pdzxp', 'This is my salt'); -- Result: 1234567
 id_decode_once 
----------------
        1234567
(1 row)

SELECT id_decode_once('PlRPdzxpR7', 'This is my salt', 10); -- Result: 1234567
 id_decode_once 
----------------
        1234567
(1 row)

SELECT id_decode_once('3GJ956J9B9', 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 1234567
 id_decode_once 
----------------
        1234567
(1 row)

